Source/ARTRealtime.m

```diff
@@ -227,7 +227,10 @@ - (void)connect {
 }
 
 - (void)_connect {
-    if(self.connection.state_nosync == ARTRealtimeClosing) {
+    NSTimeInterval intervalSinceLast = [[NSDate date] timeIntervalSinceDate:_lastActivity];
+    
+    // We want to enforce a new connection also when there hasn't been activity for longer than (idle interval + TTL)
+    if(self.connection.state_nosync == ARTRealtimeClosing || intervalSinceLast > (_maxIdleInterval + _connectionStateTtl)) {
```

Comment:
Hmm, I think that could easily create too many connections. I'll check that in more detail tomorrow.

Reply:
Confirmed, tests are failing and from the tests... 

```
[15:26:38]: ▸ ✗ Connection__connection_request_fails__should_transition_to_Failed_state_because_the_token_is_invalid_and_not_renewable, expected to equal <15>, got <50>
[15:26:38]: ▸ ✗ Connection__connection_request_fails__should_transition_to_Failed_state_because_the_token_is_invalid_and_not_renewable, Should have only one connection request fail
```

this is not good.